File: interpreter-for-cpp/src/VirtualJourney.cpp
#include "VirtualJourney.h"
#include "Runtime/Executer.h"

using namespace peak;
using namespace peak::interpreter;

VirtualJourney::VirtualJourney(std::shared_ptr<Executer> executer)
	: _executer(executer) {
}

bool VirtualJourney::Execute() {
	return _executer->Execute();
}

std::shared_ptr<Variable> VirtualJourney::FindVariable(const std::string& name) {
	return _executer->FindVariable(name);
}
bool VirtualJourney::AddVariable(std::shared_ptr<Variable> variable) {
	return _executer->AddVariable(variable);
}

File: interpreter-for-cs/src/VirtualJourney.cs
/*
	PeakScript: https://github.com/Jenocn/PeakScript
	By Jenocn: https://jenocn.github.io
*/

using System.Collections.Generic;

namespace peak {

	using interpreter;

	public class VirtualJourney {
		private Executer _executer = null;
		public VirtualJourney(Executer executer) {
			_executer = executer;
		}

		public bool Execute() {
			return _executer.Execute();
		}

		public Variable FindVariable(string name) {
			return _executer.FindVariable(name);
		}
		public bool AddVariable(Variable variable) {
			return _executer.AddVariable(variable);
		}
	}
} // namespace peak